.github/workflows/build.yml

```diff
@@ -7,40 +7,57 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  build:
+  setup:
     runs-on: ubuntu-latest
-    timeout-minutes: 10
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
       - name: Check Nix flake inputs
         uses: DeterminateSystems/flake-checker-action@v5
+      - name: Output versions
+        run: |
+          VERSIONS=$(jq -c keys_unsorted versions.json)
+          echo "versions=$VERSIONS" >> $GITHUB_OUTPUT
+        id: generate
+    outputs:
+      versions: ${{ steps.generate.outputs.versions }}
+
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    needs: [setup]
+    strategy:
+      matrix:
+        version: ${{ fromJSON(needs.setup.outputs.versions) }}
+      fail-fast: false
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
       - name: Install Nix
-        uses: DeterminateSystems/nix-installer-action@v9
+        uses: cachix/install-nix-action@v24
```

Comment:
I'm curious about this change, can you elaborate a bit?

Reply:
@oscar-izval good question, I just found that this action is slightly faster than the previous one in about 1 second for Linux, haven’t tried on macOS yet. I’m not sure if it matters a lot but now that we are using a dynamic matrix I was trying to optimize the build as much as I can.